src/robot/running/status.py

```diff
@@ -202,6 +201,7 @@ def _skip_on_failure(self):
         critical = not critical_pattern or critical_pattern.match(tags)
         skip_on_fail_pattern = TagPatterns(self._skip_on_failure_tags)
         skip_on_fail = skip_on_fail_pattern and skip_on_fail_pattern.match(tags)
+        skip_on_fail = TagPatterns("robot:skip-on-failure").match(tags)
```

Comment:
Overwriting earlier `skip_on_fail` variable certainly cannot be right. I'm surprised if there aren't tests that fail due to that. It would probably be easiest to just add `robot:skip-on-failure` at the end of `skip_on_failure_tags` either here or already in `__init__`.